open-metadata-implementation/adapters/open-connectors/integration-connectors/elasticsearch-integration-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/elasticsearch/ffdc/ElasticsearchIntegrationConnectorAuditCode.java

```diff
@@ -0,0 +1,119 @@
+     *
+     *     ElasticsearchIntegrationConnectorAuditCode   auditCode = ElasticsearchIntegrationConnectorAuditCode.CONNECTOR_STOPPING;
+     *
+     * This will expand out to the 4 parameters shown below.
+     *
+     * @param messageId - unique Id for the message
+     * @param severity - severity of the message
+     * @param message - text for the message
+     * @param systemAction - description of the action taken by the system when the condition happened
+     * @param userAction - instructions for resolving the situation, if any
+     */
+    ElasticsearchIntegrationConnectorAuditCode(String                     messageId,
+                                             OMRSAuditLogRecordSeverity severity,
+                                             String                     message,
+                                             String                     systemAction,
+                                             String                     userAction)
+    {
+        this.logMessageId = messageId;
+        this.severity = severity;
+        this.logMessage = message;
+        this.systemAction = systemAction;
+        this.userAction = userAction;
+    }
+
+
+    /**
+     * Retrieve a message definition object for logging.  This method is used when there are no message inserts.
+     *
+     * @return message definition object.
+     */
+    @Override
+    public AuditLogMessageDefinition getMessageDefinition()
+    {
+        return new AuditLogMessageDefinition(logMessageId,
+                                             severity,
+                                             logMessage,
+                                             systemAction,
+                                             userAction);
+    }
+
+
+    /**
+     * Retrieve a message definition object for logging.  This method is used when there are values to be inserted into the message.
+     *
+     * @param params array of parameters (all strings).  They are inserted into the message according to the numbering in the message text.
+     * @return message definition object.
+     */
+    @Override
+    public AuditLogMessageDefinition getMessageDefinition(String ...params)
+    {
+        AuditLogMessageDefinition messageDefinition = new AuditLogMessageDefinition(logMessageId,
+                                                                                    severity,
+                                                                                    logMessage,
+                                                                                    systemAction,
+                                                                                    userAction);
+        messageDefinition.setMessageParameters(params);
+        return messageDefinition;
+    }
+
+
+    /**
+     * JSON-style toString
+     *
+     * @return string of property names and values for this enum
+     */
+    @Override
+    public String toString()
```

Comment:
do we need equals and hashCode ?

Reply:
the class is an enum, i don't think they need to be implemented